src/prpy/planning/ompl.py

```diff
@@ -163,67 +167,115 @@ def _Plan(self, robot, goal=None, tsrchains=None, timeout=30., shortcut_timeout=
+                 algorithm='RRTConnect', robot_checker_factory=None):
+        if multiplier is None and fraction is None:
+            raise ValueError('Either "multiplier" of "fraction" is required.')
+        if multiplier is not None and fraction is not None:
+            raise ValueError('"multiplier" and "fraction" are exclusive.')
+        if multiplier is not None and not (multiplier >= 1):
+            raise ValueError('"mutiplier" must be a positive integer.')
+        if fraction is not None and not (0. <= fraction <= 1.):
+            raise ValueError('"fraction" must be between zero and one.')
+
+        OMPLPlanner.__init__(self, algorithm='RRTConnect',
+                robot_checker_factory=robot_checker_factory)
+
+        self.multiplier = multiplier
+        self.fraction = fraction
+
+
+    """ Set a default 'range' parameter from DOF resolutions.
+
+    This function returns a copy of the input 'ompl_args' argument with the
+    'range' parameter set to a default value, if it was not already set. The
+    default value is calculated from the active DOFs of 'robot' such that there
+    will be an fixed number of state validity checks per extension. That number
+    is configured in the constructor of this class.
+
+    @param robot with active DOFs set
+    @param ompl_args arguments to append to, defaults to an empty dictionary
+    @return copy of ompl_args with 'range' set to the default value
+    """
     def _SetPlannerRange(self, robot, ompl_args=None):
-        from copy import deepcopy
+        epsilon = 1e-6
 
         if ompl_args is None:
             ompl_args = dict()
         else:
             ompl_args = deepcopy(ompl_args)
 
-        # Set the default range to the collision checking resolution.
         if 'range' not in ompl_args:
-            epsilon = 1e-6
-
-            # Compute the collision checking resolution as a conservative
-            # fraction of the state space extents. This mimics the logic inside
-            # or_ompl used to set collision checking resolution.
+            # Compute the maximum DOF range, treating circle joints as SO(2).
             dof_limit_lower, dof_limit_upper = robot.GetActiveDOFLimits()
-            dof_ranges = dof_limit_upper - dof_limit_lower
+            dof_ranges = numpy.zeros(robot.GetActiveDOF())
+
+            for index, dof_index in enumerate(robot.GetActiveDOFIndices()):
+                joint = robot.GetJointFromDOFIndex(dof_index)
+                axis_index = dof_index - joint.GetDOFIndex()
+
+                if joint.IsCircular(axis_index):
+                    dof_ranges[index] = 2 * numpy.pi
+                else:
+                    dof_ranges[index] = (dof_limit_upper[index]
+                                       - dof_limit_lower[index])
+
+            # Duplicate the logic in or_ompl to convert the DOF resolutions to
+            # a fraction of the longest extent of the state space.
+            maximum_extent = numpy.max(dof_ranges)
```

Comment:
Unfortunately, that's not with the maximum extent is ... you should use `numpy.linalg.norm` instead of `numpy.max`.